src/components/SignUp/SignUp.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+
+export default class SignUp extends Component {
+  state = {
+    name: '',
+    login: '',
+    password: '',
+    repeatPasswod: ''
+  };
+
+  handleChange = name => event => {
+    this.setState({ [name]: event.target.value });
+  };
+
+  loginSubmit = () => {
+    const { name, login, password, repeatPasswod } = this.state;
+    !name || !login || !password || !repeatPasswod
```

Comment:
Simplify structure of this statement. It's very hard to follow all those ? : . It's better to use separate if statements here (Note: I'm not saying not to use ternary operator at all but you should think about code readability first)